cobc/codegen.c

```diff
@@ -8718,6 +8718,47 @@ output_ec_condition_for_handler (const enum cb_handler_type handler_type)
 	}
 }
 
+static void
+output_ec_size_handler (void)
+{
+	if (CB_EXCEPTION_ENABLE(COB_EC_SIZE_ADDRESS)) {
+		output_line("if (cob_glob_ptr->cob_exception_code == %d) "
+			"cob_fatal_exception (cob_glob_ptr->cob_exception_code);",
+			CB_EXCEPTION_CODE(COB_EC_SIZE_ADDRESS));
+	}
+	if (CB_EXCEPTION_ENABLE(COB_EC_SIZE_EXPONENTIATION)) {
+		output_line("if (cob_glob_ptr->cob_exception_code == %d) "
+			"cob_fatal_exception (cob_glob_ptr->cob_exception_code);",
+			CB_EXCEPTION_CODE(COB_EC_SIZE_EXPONENTIATION));
+	}
+	if (CB_EXCEPTION_ENABLE(COB_EC_SIZE_IMP)) {
+		output_line("if (cob_glob_ptr->cob_exception_code == %d) "
+			"cob_fatal_exception (cob_glob_ptr->cob_exception_code);",
+			CB_EXCEPTION_CODE(COB_EC_SIZE_IMP));
+	}
+	if (CB_EXCEPTION_ENABLE(COB_EC_SIZE_OVERFLOW)) {
+		output_line("if (cob_glob_ptr->cob_exception_code == %d) "
+			"cob_fatal_exception (cob_glob_ptr->cob_exception_code);",
+			CB_EXCEPTION_CODE(COB_EC_SIZE_OVERFLOW));
+	}
+	if (CB_EXCEPTION_ENABLE(COB_EC_SIZE_TRUNCATION)) {
+		output_line("if (cob_glob_ptr->cob_exception_code == %d) "
+			"cob_fatal_exception (cob_glob_ptr->cob_exception_code);",
+			CB_EXCEPTION_CODE(COB_EC_SIZE_TRUNCATION));
+	}
+	if (CB_EXCEPTION_ENABLE(COB_EC_SIZE_UNDERFLOW)) {
+		output_line("if (cob_glob_ptr->cob_exception_code == %d) "
+			"cob_fatal_exception (cob_glob_ptr->cob_exception_code);",
+			CB_EXCEPTION_CODE(COB_EC_SIZE_UNDERFLOW));
+	}
+	if (CB_EXCEPTION_ENABLE(COB_EC_SIZE_ZERO_DIVIDE)) {
+		output_line("if (cob_glob_ptr->cob_exception_code == %d) "
+			"cob_fatal_exception (cob_glob_ptr->cob_exception_code);",
+			CB_EXCEPTION_CODE(COB_EC_SIZE_ZERO_DIVIDE));
+	}
```

Comment:
can't you just have an integer `exception_code = 0` at the start of the function, then || the enabled exception values into it and then finally - if it isn't zero output one single exception check?